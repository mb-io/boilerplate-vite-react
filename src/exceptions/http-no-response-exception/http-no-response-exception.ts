import { AxiosError } from 'axios'

export class HttpNoResponseException extends Error {
  error?: AxiosError

  constructor(message?: string, error?: AxiosError) {
    super(message ?? 'Server failed to respond')
    this.error = error
    this.name = 'NoResponseException'
  }
}
