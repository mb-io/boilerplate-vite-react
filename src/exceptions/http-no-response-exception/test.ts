import { describe, it, expect } from 'vitest'
import { HttpNoResponseException } from './http-no-response-exception'
import { AxiosError } from 'axios'

describe('HttpNoResponseException', () => {
  it('should create an exception with default message', () => {
    const exception = new HttpNoResponseException()
    expect(exception.name).toBe('NoResponseException')
    expect(exception.message).toBe('Server failed to respond')
    expect(exception).toMatchSnapshot()
  })

  it('should create an exception with custom message', () => {
    const message = 'Custom no response message'
    const exception = new HttpNoResponseException(message)
    expect(exception.name).toBe('NoResponseException')
    expect(exception.message).toBe(message)
    expect(exception).toMatchSnapshot()
  })

  it('should create an exception with custom message and AxiosError', () => {
    const message = 'Custom no response message'
    const axiosError = new AxiosError('Axios error message')
    const exception = new HttpNoResponseException(message, axiosError)
    expect(exception.name).toBe('NoResponseException')
    expect(exception.message).toBe(message)
    expect(exception.error).toBe(axiosError)
    expect(exception).toMatchSnapshot()
  })
})
