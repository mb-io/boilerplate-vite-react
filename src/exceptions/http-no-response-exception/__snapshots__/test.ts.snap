// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HttpNoResponseException > should create an exception with custom message 1`] = `[NoResponseException: Custom no response message]`;

exports[`HttpNoResponseException > should create an exception with custom message and AxiosError 1`] = `[NoResponseException: Custom no response message]`;

exports[`HttpNoResponseException > should create an exception with default message 1`] = `[NoResponseException: Server failed to respond]`;
